addon/templates/components/md-nav-link.hbs

```diff
@@ -0,0 +1,6 @@
+<a href={{href}}>
+{{#if icon}}
+  <i class="material-icons {{iconClass}}">{{icon}}</i>
+{{/if}}
+{{#if linkTitle}}{{linkTitle}}{{else}}{{yield}}{{/if}}
```

Comment:
This is an effort to keep consistent with `{{link-to}}`, since this component is intended to be private (I know I could do a lot better being clear about this) and exposed as a contextual component

```hbs
{{#md-nav as |n|}}
  {{n.title 'My App' 'index'}}
  {{#n.menu class='left' as |m|}}
    {{!-- some md-nav-link components --}}
    {{m.link-to 'Thing 1' 'app.myroute' (query-params a="1") icon='star'}}
    {{m.link-to 'Thing 2' 'app.myroute' (query-params a="2")}}
  {{/n.menu}}
{{/md-nav}}
```
Under the hood, these are extended `{{link-to}}` (so we get all the good stuff that comes along with them). Unfortunately, because positional parameters are used in the base class, it would be a lot of work to support what you're asking for

What you're trying to do might feel more natural as
```hbs
{{#m.link-to 'index'}}<i class="material-icons">star</i>{{/m-link-to}}
```